core/src/main/java/edu/wpi/grip/core/sources/MultiImageFileSource.java

```diff
@@ -0,0 +1,139 @@
+
+    private EventBus eventBus;
+    private List<String> paths;
+    private int index = 0;
+
+    /**
+     * @param eventBus The event bus.
+     * @param files A list of files to be loaded.
+     * @param index The index to use as the first file that is in the socket.
+     * @throws IOException If the source fails to load any of the images
+     */
+    public MultiImageFileSource(final EventBus eventBus, final List<File> files, int index) throws IOException {
+        this.initialize(eventBus, files.stream()
+                .map(file -> URLDecoder.decode(Paths.get(file.toURI()).toString()))
+                .collect(Collectors.toList()), index);
+    }
+
+    public MultiImageFileSource(final EventBus eventBus, final List<File> files) throws IOException {
+        this(eventBus, files, 0);
+    }
+
+    /**
+     * Used only for serialization
+     */
+    public MultiImageFileSource(){ /* no op */ }
+
+
+    @SuppressWarnings("unchecked")
+    private void initialize(final EventBus eventBus, final List<String> paths, int index) throws IOException {
+        this.eventBus = checkNotNull(eventBus, "Event bus can not be null");
+        this.index = checkElementIndex(index, paths.size(), "File List Index");
+        this.paths = paths;
+        this.outputSocket = new OutputSocket(eventBus, imageOutputHint);
+        for (String path : paths) {
+            // Ensure that all of the images can be loaded
+            loadImage(path);
+        }
+        loadImage(paths.get(index));
+    }
+
+    @Override
+    public String getName() {
+        return "Multi-Image";
+    }
+
+    @Override
+    protected OutputSocket[] createOutputSockets() {
+        return new OutputSocket[]{
+                outputSocket
+        };
+    }
+
+    @Override
+    public Properties getProperties() {
+        final Properties properties = new Properties();
+        properties.setProperty(SIZE_PROPERTY, Integer.toString(paths.size()));
+        properties.setProperty(INDEX_PROPERTY, Integer.toString(index));
+        for (int i = 0; i < paths.size(); i++) {
+            properties.setProperty(getPathProperty(i), paths.get(i));
+        }
+        return properties;
+    }
+
+    @Override
+    public void createFromProperties(EventBus eventBus, Properties properties) throws IOException {
+        final int index = Integer.valueOf(properties.getProperty(INDEX_PROPERTY));
+        final int size = Integer.valueOf(properties.getProperty(SIZE_PROPERTY));
+        final List<String> paths = new ArrayList<>(size);
+        for (int i = 0; i < size; i++) {
+            paths.add(properties.getProperty(getPathProperty(i)));
+        }
+        this.initialize(eventBus, paths, index);
+    }
+
+
+    private void loadImage(String path) throws IOException {
+        OpenCVUtility.loadImage(path, this.outputSocket.getValue().get());
+        this.outputSocket.setValue(this.outputSocket.getValue().get());
+        this.eventBus.post(new SourceStartedEvent(this));
```

Comment:
What does "source started" even mean?  It seems like this method is called whenever you switch images, yet it sends a `SourceStartedEvent`.